### PR TITLE
renamed ert folder res

### PR DESCRIPTION
**Pull request by pgdr**

### CODE REVIEW

**Comment by joakim-hove** [comment]
All the Python tests need to be updated to use `ecl`- i.e. at least:
```
- from ert.test import ExtendedTestCase
+ from ecl.test import ExtendedTestCase
```

File: .travis.yml [hunk 1]
```diff
@@ -54,24 +54,29 @@ install:
...
-  - export LIBECL_BUILD_ROOT="/tmp/libecl/$TRAVIS_BUILD_ID"
-  - export LIBECL_INSTALL_ROOT="$TRAVIS_BUILD_DIR/../libecl/install"
-  - git clone https://github.com/Statoil/libecl $LIBECL_BUILD_ROOT
-  - mkdir $LIBECL_BUILD_ROOT/build
-  - pushd $LIBECL_BUILD_ROOT/build
+  - pushd ..
+  - git clone https://github.com/Statoil/libecl
+  - mkdir libecl/build
+  - pushd libecl/build
+  - export LIBECL_INSTALL_ROOT=`pwd`/install
```
Comment (joakim-hove) [inline .travis.yml]: Instead of backtics you can use: `$(pwd)/install` - which is in my opinion cleaner.

File: python/CMakeLists.txt [hunk 2]
```diff
@@ -1,68 +1,35 @@
...
-set( PY_VERSION 2.7 )
-FIND_PACKAGE(PythonInterp ${PY_VERSION} EXACT)
-if (NOT DEFINED PYTHON_EXECUTABLE)
-   message(WARNING "Python${PY_VERSION} interpreter not found - Python wrappers not enabled")
-   return()
-endif()
+init_python(2.7)
 
-set(PYTHON_INSTALL_PREFIX  "lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/${PYTHON_PACKAGE_PATH}"  CACHE STRING "Subdirectory to install Python modules in")
+set(PYTHON_INSTALL_PREFIX
```
Comment (joakim-hove) [inline python/CMakeLists.txt]: This will by set by the `init_python( )` macro.

File: python/python/res/config/__init__.py [hunk 3]
```diff
@@ -14,11 +14,11 @@
 #  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
 #  for more details.
 from cwrap import Prototype
-import ert
+import ecl
 import ecl.util
 
 class ConfigPrototype(Prototype):
-    lib = ert.load("libconfig")
+    lib = ecl.load("libconfig")
```
Comment (joakim-hove) [inline python/python/res/config/__init__.py]: This should be `res.load( )`?

File: python/python/res/enkf/__init__.py [hunk 4]
```diff
@@ -27,14 +27,14 @@
 import res.job_queue
 
 class EnkfPrototype(Prototype):
-    lib = ert.load("libenkf")
```
Comment (joakim-hove) [inline python/python/res/enkf/__init__.py]: `res.load( )`?

File: python/tests/res/analysis/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    test_linalg.py
+    test_options_enum.py
+    test_rml.py
+    test_std_enkf.py
+    test_std_enkf_debug.py
+)
+
+add_python_package("python.tests.res.analysis" ${PYTHON_INSTALL_PREFIX}/tests/res/analysis "${TEST_SOURCES}" False)
+
+addPythonTest(res.analysis.analysis_module tests.res.analysis.test_analysis_module.AnalysisModuleTest)
```
Comment (joakim-hove) [inline python/tests/res/analysis/CMakeLists.txt]: All calls to `addPythonTest( )` should be updated to the new version of the function (found `libecl`) - which only takes the Python package path to the test class as argument.
```bash
addPythonTest(tests.res.analysis.test_analysis_module.AnalysisModuleTest)
```

File: python/tests/CMakeLists.txt [hunk 6]
```diff
@@ -32,11 +32,11 @@ endfunction(addPythonTest)
...
 
 if(BUILD_ERT)
-    add_subdirectory(ert)
+    add_subdirectory(res)
 
-    add_test( NAME python.tests.ert.import
+    add_test( NAME python.tests.res.import
               WORKING_DIRECTORY ${INSTALL_DIRECTORY}
-              COMMAND tests/ctest_import.py ${INSTALL_DIRECTORY} "ert")
+              COMMAND tests/ctest_import.py ${INSTALL_DIRECTORY} "res")
```
Comment (joakim-hove) [inline python/tests/CMakeLists.txt]: Look at teh global tests in [`libecl`](https://github.com/Statoil/libecl/blob/master/python/tests/global/CMakeLists.txt)

Comment (joakim-hove) [inline python/tests/CMakeLists.txt]: The `addPythonTest( )` function definition from this file should be removed - the `init_python( )` macro should ensure that we use the version found in `libecl`.

File: python/CMakeLists.txt [hunk 7]
```diff
@@ -1,68 +1,35 @@
 include(cmake/find_python_module.cmake)
```
Comment (joakim-hove) [inline python/CMakeLists.txt]: These include statements should be removed (and the files which are included also removed) - this comes via `init_python( )` from `libecl`.

**Comment by joakim-hove** [comment]
All the python related files under `${ROOT}/cmake` should be removed - these should now be provided from `libecl`.

File: python/python/ert/test/test_run.py [hunk 8]
```diff
@@ -1,158 +0,0 @@
-#  Copyright (C) 2013  Statoil ASA, Norway.
```
Comment (joakim-hove) [inline python/python/ert/test/test_run.py]: This file can live - in `res/test/test_run.py`

File: .travis.yml (hunk 1 above)
Comment (pgdr) [inline .travis.yml]: You are absolutely right.  Plain old nostalgia on my part.  Will fix.

File: python/CMakeLists.txt (hunk 2 above)
Comment (pgdr) [inline python/CMakeLists.txt]: Will remove.

File: python/python/res/config/__init__.py (hunk 3 above)
Comment (pgdr) [inline python/python/res/config/__init__.py]: You are right; should add tests as well?

File: python/tests/res/analysis/CMakeLists.txt (hunk 5 above)
Comment (pgdr) [inline python/tests/res/analysis/CMakeLists.txt]: Will update.

File: python/tests/CMakeLists.txt (hunk 6 above)
Comment (pgdr) [inline python/tests/CMakeLists.txt]: You are right.  I will remove them.

File: python/CMakeLists.txt (hunk 7 above)
Comment (pgdr) [inline python/CMakeLists.txt]: Will remove.

**Comment by pgdr** [comment]
Thanks for the review, @joakim-hove.  I will address the comments tomorrow morning (when I get paid for addressing them).

File: python/python/res/config/__init__.py (hunk 3 above)
Comment (pgdr) [inline python/python/res/config/__init__.py]: Oh, no, this, I believe should be `ecl`.  The `load` function from `ecl` is the one defined from `cwrap`.

libecl still contains this functionality, and it does not exist in libres.

**Pull request merged**